### PR TITLE
Add testing for drag and drop events/behavior on `QtLayerList`

### LINKED ISSUE #1411: Better Qt tests for mouse move/ drag events
## 🐛 Bug

We have custom mouse move/ drag events on some of our qt objects like the `QtLayerList`, but we don't have any good testing yet of these functions - see https://codecov.io/gh/napari/napari/compare/e43f730170aab1e011c42060c65ba44133f6f18f...87910700153881903401c4f1d3f58285c94a674b/src/napari/_qt/qt_layerlist.py?before=napari/_qt/qt_layerlist.py from #1391. We should add dedicated tests for these lines and other similar ones that simulate Qt mouse moves.

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

Closes #1411 

# Description

Add a test for the layer list drag and drop behavior (use `pyautogui` and the `skip_local_focus` decorator):

dragdrop

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
@dalthviz can you clarify, when running locally with `NAPARI_FOCUS_TESTS=1` do I need to manually drag the layers (it passes when I do) or should pyautogui be taking care of that?

**Comment by dalthviz** [comment]
Hi! It should be done automatically (the mouse mouvement and drag&drop action over the widget should be done without you needing to do any manual action) 🤔 If you are checking locally on a machine with macOS, could you give the terminal app `Accessibility` permissions? You can give it permissions by going to `System Settings > Privacy & Security > Accessibility` Let me know if that helps!

**Comment by psobolewskiPhD** [comment]
That did the trick! Pretty amazing! 
I think we'll need to document this somewhere, maybe here:
https://napari.org/stable/developers/testing.html#running-tests

(BTW that seems wrong doesn't it? 
> There are a very small number of tests (<5) that require showing GUI elements, (such as testing screenshots). By default, these are only run during continuous integration. If you’d like to include them in local tests, set the environment variable “CI”

I know I get windows showing when running locally and it's more than 5... Maybe I'm misunderstanding)

**Comment by dalthviz** [comment]
> That did the trick! Pretty amazing!

Awesome :tada:

> I think we'll need to document this somewhere, maybe here:
> https://napari.org/stable/developers/testing.html#running-tests

That makes sense and I think I will add a comment over the test docstring to clarify that there too :+1: Also, could make sense to add in the docs the usage of `skip_local_focus` following the phrasing at https://napari.org/dev/developers/contributing/testing.html#skipping-tests-that-show-gui-elements (either in its own section or as part of that section)?

> (BTW that seems wrong doesn't it?
> 
>     There are a very small number of tests (<5) that require showing GUI elements, (such as testing screenshots). By default, these are only run during continuous integration. If you’d like to include them in local tests, set the environment variable “CI”
> 
> I know I get windows showing when running locally and it's more than 5... Maybe I'm misunderstanding)

I think some changes have been done to update that at https://github.com/napari/docs/pull/370 so if you check the page from the dev/latest version of the docs (https://napari.org/dev/developers/contributing/testing.html#running-tests) you see a more up to date phrasing like:

     There are some tests that require showing GUI elements, (such as testing screenshots). By default, these are only run during continuous integration...

Which I think reflects the current test suite state regarding showing GUI elements when running it (indeed more than 5 tests showing GUI elements)

File: napari/_qt/containers/_tests/test_qt_layer_list.py [hunk 1]
```diff
@@ -191,6 +193,60 @@ def make_qt_layer_list_with_delegate(qtbot):
     return image1, image2, image3, layers, view, delegate
 
 
+@skip_local_focus
+def test_drag_and_drop_layers(qtbot):
+    """
+    Test drag and drop actions with pyautogui to change layer list order.
+
+    Notes:
+        * For this test to pass locally, you need to give the Terminal/iTerm
```
Comment (psobolewskiPhD) [inline napari/_qt/containers/_tests/test_qt_layer_list.py]:
```suggestion
        * For this test to pass locally on macOS, you need to give the Terminal/iTerm
```
Not sure what the situation would be for other platforms? Does it JustWork?

**Review by psobolewskiPhD — APPROVED** [review]
Love it. Makes sense and I checked it locally too -- the new docstring is great, one tiny nitpick/question.
I made a few comments to the partner docs PR -- thanks for taking care of that!

File: napari/_qt/containers/_tests/test_qt_layer_list.py (hunk 1 above)
Comment (dalthviz) [reply napari/_qt/containers/_tests/test_qt_layer_list.py]: Actually that's a good question :+1: as far as I checked on Windows seems like things just work, on Linux is the same although when running the tests over the CI a need for an `.Xauthority` file raised (which is being taken care with a validation at the `conftest.py` file thanks to https://github.com/napari/napari/pull/6719).

I will apply your suggestion but let me know if putting further info in the comment could be worthy!

**Comment by psobolewskiPhD** [comment]
But what python version?
as of
https://github.com/napari/napari/pull/5244
we don't allow 6.3.2 on python 3.10 or above.

**Comment by dalthviz** [comment]
> But what python version?

I was checking with Python 3.10 indeed

>as of
https://github.com/napari/napari/pull/5244
we don't allow 6.3.2 on python 3.10 or above.

Thanks for linking the place where the constraint was done (I was actually unsure why the tests were constrained to that specific PySide6 version on Python 3.9 :+1:)

**Review by DragaDoncila — APPROVED** [review]
Sorry this one fell through the cracks @dalthviz! Given that it's been previously approved and CI is green, I will mark ready-to-merge. @psobolewskiPhD @Czaki please take a look and let us know what needs to be done here if you don't think this is ready.

**Pull request merged**